pkg/cmd/pulumi-language-java/main.go

```diff
@@ -177,7 +177,10 @@ func (host *javaLanguageHost) determinePulumiPackages(
 	args := host.exec.pluginArgs
 	output, err := host.runJavaCommand(ctx, cmd, args)
 	if err != nil {
-		return nil, errors.Wrapf(err, "language host could not run plugin discovery command successfully")
+		// Plugin determination is an advisory feature so it doe not need to escalate to an error.
+		logging.V(3).Infof("language host could not run plugin discovery command successfully, returning empty plugins; cause: %s", err)
+		logging.Warningf("Plugin autodiscovery failed. More information available at log level 3 and above")
```

Comment:
@t0yv0 I've added a warning, if you are sure we don't need it I'll remove it, but I wanted to avoid a situation where the plugin discovery fails and the user is confused why this blows up in runtime on a missing plugin, esp. important when there are no cached plugins on the users machine yet

Reply:
Would it actually blow up on a missing plugin? My understanding is that it will still auto-install it at a later phase which is less desirable but still ok. Also what about programs that do not need plugins at all?

I feel that we should not confuse the user with the error message here, since the message makes it sound like a Pulumi error and does not point to how to resolve it.

In the vast majority of cases we hit this pathway there is a problem with the project, either the code does not compile or builder cannot resolve dependencies or some such thing. The message we show the user should convey that - even if it's unannotated builder output with the build command we ran so the user can reproduce. Since we don't parse the builder output we don't know at the Go level any more detail about the error.

I continue to be inclined this error message should be coming out of the main build pathway not the error discovery pathway (where I'd prefer just hiding things in INFO level), but I'm less firm on that; if the error message is not confusing it can come from here.

Reply:
thank you, I've removed the warnings